ibis/backends/pandas/tests/test_core.py

```diff
@@ -50,52 +47,22 @@ def test_from_dataframe(dataframe, ibis_table, core_client):
     tm.assert_frame_equal(result, expected)
 
 
-def test_pre_execute_basic():
-    """Test that pre_execute has intercepted execution and provided its own
-    scope dict."""
-
-    @pre_execute.register(ops.Add)
-    def pre_execute_test(op, *clients, scope=None, **kwargs):
-        return Scope({op: 4}, None)
-
-    one = ibis.literal(1)
-    expr = one + one
-    result = execute(expr.op())
-    assert result == 4
-
-    del pre_execute.funcs[(ops.Add,)]
-    pre_execute.reorder()
-    pre_execute._cache.clear()
-
-
 def test_execute_parameter_only():
     param = ibis.param("int64")
-    result = execute(param.op(), params={param.op(): 42})
+    con = ibis.pandas.connect()
+    result = con.execute(param, params={param.op(): 42})
     assert result == 42
 
 
 def test_missing_data_sources():
     t = ibis.table([("a", "string")])
     expr = t.a.length()
-    with pytest.raises(com.UnboundExpressionError):
-        execute(expr.op())
-
-
-def test_post_execute_called_on_joins(dataframe, core_client, ibis_table):
-    count = [0]
-
-    @post_execute.register(ops.InnerJoin, pd.DataFrame)
-    def tmp_left_join_exe(op, lhs, **kwargs):
-        count[0] += 1
-        return lhs
-
-    left = ibis_table
-    right = left.view()
-    join = left.join(right, "plain_strings")[left.plain_int64]
-    result = join.execute()
-    assert result is not None
-    assert not result.empty
-    assert count[0] == 1
+    con = ibis.pandas.connect()
+    with pytest.raises(com.OperationNotDefinedError):
+        con.execute(expr)
+    # TODO(kszucs): it should raise an unbound expr error
```

Comment:
Does this need to be addressed?

Reply:
Fixed, also added support for execution unbound tables with the backend.